### PR TITLE
CAMS-739 closed cases visibility

### PR DESCRIPTION
**Pull request by mattstankey**
# Purpose

Added visibility to closed cases on Case Search results screen.

As a USTP staff member, in order to avoid missing cases that may be closed, I want clear notification when closed cases match my search criteria.

Screenshot 2026-05-01 at 4 07 50 PM

# Definition of Done:

- [ ] Code refactored for clarity: Developers can understand the work simply by reviewing the code
- [ ] Dependency rule followed: More important code doesn’t directly depend on less important code
- [ ] Development debt eliminated: UX and code aligns to the team’s latest understanding of the domain
- [ ] No regressions: Changes do not cause regression in related or unrelated areas of the application

## Summary by Sourcery

Improve search results visibility for closed cases and support displaying open/closed status in case search results.

New Features:
- Show contextual alerts and hints when closed cases may match search filters but are excluded from current results, including count-specific messaging for case-number searches.
- Allow users to quickly include closed cases via an inline control in alerts that updates the search predicate and reruns the search.
- Add an optional Open/Closed column in the case search results table to indicate each case’s status.

Enhancements:
- Expose a closedCasesCount field in pagination responses when searching by case number with closed cases excluded, and surface it in the UI to drive closed-case messaging.
- Update search validators to accept Chapter 9 and adjust chapter validation messaging accordingly.
- Simplify alert styling by relying on the standard slim variant and removing the custom compact-with-title class.
- Refine mock search API behavior to better simulate closed/open cases and filter behavior for chapters and debtor name.

Tests:
- Extend backend controller tests to cover closedCasesCount population and edge cases.
- Add comprehensive UI tests for new closed-case alerts, Include Closed Cases interactions, Open/Closed column rendering, and search form ordering.
- Update Alert component tests to reflect the new slim alert behavior.

### CODE REVIEW

File: user-interface/src/search-results/SearchResults.tsx [hunk 1]
```diff
@@ -32,37 +32,73 @@ export type SearchResultsHeaderProps = {
   labels: string[];
   phoneticSearchEnabled?: boolean;
   showDebtorNameColumn?: boolean;
+  showOpenClosedColumn?: boolean;
 };
 
 export type SearchResultsRowProps = TableRowProps & {
   idx: number;
-  bCase: CaseSummary;
+  bCase: DxtrCase;
```
Comment (btposey) [inline user-interface/src/search-results/SearchResults.tsx]: Why did we use `DxtrCase`? Search results all come from Cosmos and we only store `CaseSummary` in Cosmos. `DxtrCase` for one, screams a dependency model that belongs in a gateway, not in the CAMS domain, and two if additional attributes are needed then they should be sync'ed into Cosmos and added to `CaseSummary` if included in search results. Needs some discussion here if the problem is larger or more complicated than this.

File: user-interface/src/index.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import { BrowserRouter } from 'react-router-dom';
 import './index.css';
+import './App.scss';
 import App from './App';
```
Comment (btposey) [inline user-interface/src/index.tsx:6]: Why was App.scss moved from App.tsx to the index? Was something added to App.scss that should have been added to index.css?

File: common/src/api/pagination.ts [hunk 3]
```diff
@@ -13,4 +13,5 @@ export type Pagination = {
   currentPage: number;
   totalPages?: number;
   totalCount?: number;
+  closedCasesCount?: number;
```
Comment (btposey) [inline common/src/api/pagination.ts]: There was a purity to the pagination model that this seems to polute. Pagination is not generic anymore as it contains a count specifically for the case domain.

**Review by btposey — COMMENTED** [review]
I have questions....

File: user-interface/src/index.tsx (hunk 2 above)
Comment (fmaddenflx) [reply user-interface/src/index.tsx:6]: It has to do with the order in which the build process in the new vite adds the css files.  Apparently we can't control the order.  Our css was then being overridden by USWDS but USWDS should always go in first.  This seemed to be the way to gaurantee USWDS makes it into the output before anything else.  I didn't like moving it to index.tsx but claude recommended it and after some thought, it made sense.

File: common/src/api/pagination.ts (hunk 3 above)
Comment (fmaddenflx) [reply common/src/api/pagination.ts]: Oh yeah!  This isn't good!

File: user-interface/src/search-results/SearchResults.tsx (hunk 1 above)
Comment (mattstankey) [reply user-interface/src/search-results/SearchResults.tsx]: Good catch, we shouldn't be using `DxtrCase`.  SearchResultsRowProps.bCase should stay typed as `SyncedCase` (the actual domain type used throughout the stack), not `DxtrCase`. The original change to 
  `DxtrCase` was unnecessarily reaching down the type hierarchy when the actual data is `SyncedCase`. CaseSummary was too narrow (no closedDate), but `DxtrCase` is also wrong — SyncedCase is the right type.

File: common/src/api/pagination.ts (hunk 3 above)
Comment (mattstankey) [reply common/src/api/pagination.ts]: We're going to fix this by extending `Pagination` in the cases domain — Define a `CasesPagination` = Pagination & { closedCasesCount?: number } in the cases domain (e.g., common/src/cams/) and use it only in the cases controller and search response. `Pagination` stays pure.

**Review by btposey — APPROVED** [review]
Pending agreed upon changes.
